plugins/global-search/src/utils/filter/AsyncProcessor.ts

```diff
@@ -80,33 +78,52 @@ export class TimeBasedAsyncProcessor<TInput, TOutput> {
         this.abortController = abortController
 
         const results: TOutput[] = []
-        let currentIndex = 0
+        let processedItems = 0
+        let lastProcessedKey: string | null = null
 
         try {
-            while (currentIndex < items.length) {
+            // There is an `return` in the loop, that makes sure to exit the loop when the iterator is exhausted
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+            while (true) {
```

Comment:
I'm not very happy about the `while(true)` here, but have a hard time thinking on something better. Moving the iterator out of here doesn't work, because we have to keep the reading transactions in tact.